### DISCUSSION TITLE
phase_param functions

### DISCUSSION THREAD

**Discussion by abhishek-appana** [pybamm-team/PyBaMM#3860] [Q&A] [answered]
Hello everyone,

I was going through the submodels for the DFN lithium-ion battery model. There are multiple submodels that use the function `phase_param`.

For example, 'single_ocp.py' uses `self.phase_param.U(sto_surf, T)`
'base_interface.py' uses `phase_param.j0(c_e, c_s_surf, T)`

I was not able to figure out where and how these functions are created and defined. Can someone point me to this?

It is probably something straightforward and I am not able to see it. Appreciate the help!

**Comment by martinjrobins** [comment, marked as answer]
Hi @abhishek-appana, the functions `j0` and `U` are defined in the parameters for your model, for example for the lithium ion models: https://github.com/pybamm-team/PyBaMM/blob/dfdc376706f53442314b27cf580b0c7d3a86e3e2/pybamm/parameters/lithium_ion_parameters.py#L578

**Reply by abhishek-appana** [reply]
Got it. Thank you @martinjrobins
